### PR TITLE
Adding CNV chart with CR and Operator

### PR DESCRIPTION
**Pull request by jordigilh**
Combines the CNV subscription and hyperconverged CR into a single chart. The chart contains a job whose responsibility is to delay the deployment of the hyperconverged CR until the CSV has reached the Successful state. This is to avoid the situation where the CR is created before the CRD.

@sabre1041 @nasx please review at your discretion.

### CODE REVIEW

**Comment by sabre1041** [comment]
@jordigilh we might be able to omit the need for a job by using this configuration in the `resourceCustomizations` section of the `ArgoCD` resource

```
    operators.coreos.com/Subscription:
      health.lua: |
        hs = {}
        if obj.status ~= nil then
          if obj.status.currentCSV ~= nil and (obj.status.state == "AtLatestKnown" or obj.status.state == "UpgradeAvailable" or obj.status.state == "UpgradePending") then
            hs.status = "Healthy"
            hs.message = "Subcription installed"
            return hs
          end
        end
        hs.status = "Progressing"
        hs.message = "Waiting for Subscription to complete."
        return hs
```

**Comment by jordigilh** [comment]
> @jordigilh we might be able to omit the need for a job by using this configuration in the `resourceCustomizations` section of the `ArgoCD` resource
> 
> ```
>     operators.coreos.com/Subscription:
>       health.lua: |
>         hs = {}
>         if obj.status ~= nil then
>           if obj.status.currentCSV ~= nil and (obj.status.state == "AtLatestKnown" or obj.status.state == "UpgradeAvailable" or obj.status.state == "UpgradePending") then
>             hs.status = "Healthy"
>             hs.message = "Subcription installed"
>             return hs
>           end
>         end
>         hs.status = "Progressing"
>         hs.message = "Waiting for Subscription to complete."
>         return hs
> ```

Works great! thanks!. I've removed the job and it's resources and updated the [argocd.yaml](https://github.com/redhat-cop/rhel-edge-automation-arch/blob/653fbc49aa44b30286152cf57c240b3c5eaec518/openshift/gitops/manifests/bootstrap/argocd/base/argocd.yaml#L40-L52) file to include the `resourceCustomizations`.

I've tested it without the helm pre-hooks and works fine as well. So they're gone :smile:

File: openshift/gitops/manifests/bootstrap/argocd/base/argocd.yaml [hunk 1]
```diff
@@ -37,3 +37,17 @@ spec:
       ignoreDifferences: |
         jsonPointers:
         - /spec/template/spec/containers/0/image
+    operators.coreos.com/Subscription:
+      health.lua: |
+        hs = {}
+        if obj.status ~= nil then
+          if obj.status.currentCSV ~= nil and (obj.status.state == "AtLatestKnown" or obj.status.state == "UpgradeAvailable" or obj.status.state == "UpgradePending") then
+            hs.status = "Healthy"
+            hs.message = "Subcription installed"
```
Comment (sabre1041) [inline openshift/gitops/manifests/bootstrap/argocd/base/argocd.yaml:46]:
```suggestion
            hs.message = "Subscription installed"
```

**Review by sabre1041 — CHANGES_REQUESTED** [review]

File: charts/cnv/templates/subscription.yaml [hunk 2]
```diff
@@ -0,0 +1,10 @@
+apiVersion: operators.coreos.com/v1alpha1
+kind: Subscription
+metadata:
+  name: hco-operatorhub
+  namespace: {{ .Values.namespace }}
```
Comment (sabre1041) [inline charts/cnv/templates/subscription.yaml:5]: ` namespace: {{ template "common.names.namespace" $ }}`

File: charts/cnv/templates/operatorgroup.yaml [hunk 3]
```diff
@@ -0,0 +1,8 @@
+apiVersion: operators.coreos.com/v1
+kind: OperatorGroup
+metadata:
+  name: kubevirt-hyperconverged-group
+  namespace: {{ .Values.namespace }}
```
Comment (sabre1041) [inline charts/cnv/templates/operatorgroup.yaml:5]: ` namespace: {{ template "common.names.namespace" $ }}

File: charts/cnv/templates/kubevirt-hyperconverged.yaml [hunk 4]
```diff
@@ -1,12 +1,13 @@
 apiVersion: hco.kubevirt.io/v1beta1
 kind: HyperConverged
 metadata:
-  annotations:
-    deployOVS: {{ .Values.cnv.deploy_ovs | quote }}
   name: kubevirt-hyperconverged
-  namespace: {{ template "common.names.namespace" $ }}
+  namespace: {{ .Values.namespace }}
```
Comment (sabre1041) [inline charts/cnv/templates/kubevirt-hyperconverged.yaml:5]: `namespace: {{ template "common.names.namespace" $ }}`

**Review by sabre1041 — CHANGES_REQUESTED** [review]

File: openshift/gitops/manifests/bootstrap/argocd/base/argocd.yaml (hunk 1 above)
Comment (jordigilh) [reply openshift/gitops/manifests/bootstrap/argocd/base/argocd.yaml]: Updated.

**Comment by sabre1041** [comment]
@jordigilh This is really exciting! It worked great. we most likely want to add retries, but that can come later. a few small minor changes

File: charts/cnv/templates/kubevirt-hyperconverged.yaml (hunk 4 above)
Comment (jordigilh) [reply charts/cnv/templates/kubevirt-hyperconverged.yaml]: done

File: charts/cnv/templates/operatorgroup.yaml (hunk 3 above)
Comment (jordigilh) [reply charts/cnv/templates/operatorgroup.yaml]: Done

File: charts/cnv/templates/subscription.yaml (hunk 2 above)
Comment (jordigilh) [reply charts/cnv/templates/subscription.yaml]: Done and removed the `namespace` field in the `values.yaml` file.

**Review by sabre1041 — APPROVED** [review]
LGTM!

**Pull request merged**